xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/notification/email/default.plain.vm

```diff
@@ -31,12 +31,15 @@ $app:
 ### Display a link to the diff
 ###
 #if ($document)
-#set ($newdoc = $xwiki.getDocument($event.document, $event.events[0].documentVersion))
-#if ($event.events.size() == 1)
-#set ($origdoc = $newdoc.previousDocument)
-#else
-#set ($lastIndex = $event.events.size() - 1)
-#set ($origdoc = $xwiki.getDocument($event.document, $event.events[$lastIndex].documentVersion).previousDocument)
-#end
-  $services.localization.render('notifications.email.seeChanges', [$document.getURL('view', "viewer=changes&rev1=${origdoc.version}&rev2=${newdoc.version}")])
+  #set ($newdoc = $xwiki.getDocument($event.document, $event.events[0].documentVersion))
+  #if ($event.events.size() == 1)
+    #set ($origdoc = $newdoc.previousDocument)
+  #else
+    #set ($lastIndex = $event.events.size() - 1)
+    #set ($origdoc = $xwiki.getDocument($event.document, $event.events[$lastIndex].documentVersion).previousDocument)
+  #end
+  #if ($event.type != 'delete')
+    #set ($queryString = "viewer=changes&rev1=${origdoc.version}&rev2=${newdoc.version}")
+  #end
```

Comment:
What happens to `$queryString` when `$event.type == 'delete'`? Can it contain some content from a previous usage? I would be more secure of `$queryString` were initialized first.

Reply:
BTW, thanks for the Pull Request :)